reporter/src/main/kotlin/reporters/AmazonOssAttributionBuilderReporter.kt

```diff
@@ -0,0 +1,194 @@
+    override val reporterName = "AmazonOssAttributionBuilder"
+    override val defaultFilename = "OssAttribution.txt"
+
+    private val service = AmazonOssAttributionBuilderService.create(
+        AmazonOssAttributionBuilderService.Server.DEFAULT,
+        OkHttpClientHelper.buildClient(HTTP_CACHE_PATH)
+    )
+
+    override fun generateReport(outputStream: OutputStream, input: ReporterInput) {
+        // TODO: Allow to configure username and password via the config file; use the defaults for now.
+        val credentials = Credentials.basic("admin", "admin")
+
+        val rootProject = input.ortResult.getProjects().singleOrNull()
+
+        // TODO: Add support for multiple projects.
+        requireNotNull(rootProject) {
+            "The ORT result must contain exactly one project."
+        }
+
+        try {
+            // Create a new project which gives all permissions to everyone for simplicity.
+            val acl = mapOf("everyone" to AmazonOssAttributionBuilderService.Role.OWNER)
+
+            // Projects have a list of contacts, but at the moment the UI only supports one contact, the legal contact.
+            val contacts = AmazonOssAttributionBuilderService.ProjectContacts(listOf("Insert legal contact here."))
+
+            val metadata = jsonMapper.createObjectNode().put("open_sourcing", true)
+            val newProject = AmazonOssAttributionBuilderService.NewProject(
+                rootProject.id.name,
+                rootProject.id.version,
+                "Imported from results of the OSS Review Toolkit.",
+                Instant.now().toString(), // Use a fake planned release date.
+                contacts,
+                acl,
+                metadata
+            )
+
+            val newProjectResponse = service.createNewProject(newProject, credentials).execute()
+            val newProjectBody = getBodyOrHandleError(newProjectResponse, "Unable to create a new project")
+
+            log.info { "Successfully created project with id '${newProjectBody.projectId}'." }
+
+            // Attach packages to the newly created project.
+            val dependencies = input.ortResult.collectDependencies(rootProject.id)
+            dependencies.forEach { id ->
+                // We know that a package exists for the reference.
+                val pkg = input.ortResult.getPackage(id)!!.pkg
+
+                // TODO: Theoretically, a project could use the same package multiple times in different linkages in the
+                //       ORT model. Think about how to map that to the Attributon Builder.
+                //       If the package was modified is something that is not currently capture by ORT. Think about
+                //       whether this is something we should do.
+                val usage = AmazonOssAttributionBuilderService.Usage("No notes.", link = "dynamic", modified = false)
+
+                val pkgUrl = pkg.homepageUrl.takeUnless { it.isEmpty() } ?: "https://github.com/404"
```

Comment:
Because the URL being passed here actually *needs* to be reachable (!), also see [this comment](https://github.com/oss-review-toolkit/ort/pull/2467/files/8ecc9c019442b6e3498390e09091682c588b15b2#diff-a41433a03d09e22b438c5cef6d496d0eR122).

Reply:
Can you add this as a code comment above this line?

Reply:
Please either write this like

    // URL passed into the Amazon Oss Attribution Builder needs (!) to be reachable.

or

    // URL passed into the Amazon Oss Attribution Builder needs to be reachable!